### ISSUE TITLE
[BUG] Driver deployer stuck in Init / UI in CrashLoopBackOff / Manager cannot list the content of the src directory /var/lib/rancher/longhorn/engine-binaries

### ISSUE THREAD

**Issue by FabianNiehaus** [longhorn/longhorn#2147]
**Describe the bug**
When installing Longhorn via Helm in a new cluster, the UI and the driver-deployer are not started properly, and the managers raise errors in the log.

**To Reproduce**
1. Install Longhorn v1.0.0 via Helm 3
    ```bash
   kubectl create namespace longhorn-system
   helm install longhorn longhorn/longhorn --namespace longhorn-system --values longhorn-values.yaml
   ```
2. Wait for Pods to come online
   ```bash
   watch -n 1 kubectl -n longhorn-system get pod 
   ```
4. Pods get stuck at this state:
    ```
    NAME                                        READY   STATUS             RESTARTS   AGE
    longhorn-driver-deployer-76c8fd69fb-hqcwk   0/1     Init:0/1           0          20m
    instance-manager-e-bc8f8398                 1/1     Running            0          20m
    engine-image-ei-cf743f9c-875pd              1/1     Running            0          20m
    engine-image-ei-cf743f9c-cp7xw              1/1     Running            0          20m
    instance-manager-e-06171c88                 1/1     Running            0          20m
    longhorn-manager-lsxqm                      1/1     Running            1          20m
    longhorn-manager-98lqf                      1/1     Running            2          20m
    longhorn-ui-78547884d8-dpqjq                0/1     CrashLoopBackOff   8          20m
    ```

**Expected behavior**
All Pods come online without error.

**Log**
```
time="2021-01-02T16:02:02Z" level=info msg="Start overwriting built-in settings with customized values"
W0102 16:02:02.965614       1 client_config.go:541] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
time="2021-01-02T16:02:02Z" level=info msg="cannot list the content of the src directory /var/lib/rancher/longhorn/engine-binaries for the copy, will do nothing: Failed to execute: nsenter [--mount=/host/proc/1/ns/mnt --net=/host/proc/1/ns/net bash -c ls /var/lib/rancher/longhorn/engine-binaries/*], output , stderr, ls: cannot access '/var/lib/rancher/longhorn/engine-binaries/*': No such file or directory\n, error exit status 2"
I0102 16:02:02.971349       1 leaderelection.go:241] attempting to acquire leader lease  longhorn-system/longhorn-manager-upgrade-lock...
I0102 16:02:02.982228       1 leaderelection.go:251] successfully acquired lease longhorn-system/longhorn-manager-upgrade-lock
time="2021-01-02T16:02:02Z" level=info msg="Start upgrading"
time="2021-01-02T16:02:02Z" level=info msg="No API version upgrade is needed"
time="2021-01-02T16:02:03Z" level=info msg="Finish upgrading"
time="2021-01-02T16:02:03Z" level=info msg="Upgrade leader lost: k8s-worker-01"
W0102 16:02:03.016634       1 client_config.go:541] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
I0102 16:02:03.019036       1 shared_informer.go:197] Waiting for caches to sync for longhorn datastore
I0102 16:02:03.128614       1 shared_informer.go:204] Caches are synced for longhorn datastore
W0102 16:02:03.128692       1 client_config.go:541] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.
time="2021-01-02T16:02:03Z" level=info msg="Start Longhorn replica controller"
I0102 16:02:03.135811       1 shared_informer.go:197] Waiting for caches to sync for longhorn replicas
I0102 16:02:03.135818       1 shared_informer.go:204] Caches are synced for longhorn replicas
time="2021-01-02T16:02:03Z" level=info msg="Start Longhorn engine controller" controller=longhorn-engine node=k8s-worker-01
I0102 16:02:03.135866       1 shared_informer.go:197] Waiting for caches to sync for longhorn engines
I0102 16:02:03.135869       1 shared_informer.go:204] Caches are synced for longhorn engines
time="2021-01-02T16:02:03Z" level=info msg="Start Longhorn volume controller" controller=longhorn-volume node=k8s-worker-01
I0102 16:02:03.135931       1 shared_informer.go:197] Waiting for caches to sync for longhorn engines
I0102 16:02:03.135934       1 shared_informer.go:204] Caches are synced for longhorn engines
time="2021-01-02T16:02:03Z" level=info msg="Start Longhorn Engine Image controller"
I0102 16:02:03.135962       1 shared_informer.go:197] Waiting for caches to sync for longhorn engine images
I0102 16:02:03.135971       1 shared_informer.go:204] Caches are synced for longhorn engine images
time="2021-01-02T16:02:03Z" level=info msg="Start Longhorn node controller"
I0102 16:02:03.136102       1 shared_informer.go:197] Waiting for caches to sync for longhorn node
I0102 16:02:03.136108       1 shared_informer.go:204] Caches are synced for longhorn node
time="2021-01-02T16:02:03Z" level=info msg="Start Longhorn websocket controller" controller=longhorn-websocket node=k8s-worker-01
I0102 16:02:03.136200       1 shared_informer.go:197] Waiting for caches to sync for longhorn websocket
I0102 16:02:03.136203       1 shared_informer.go:204] Caches are synced for longhorn websocket
time="2021-01-02T16:02:03Z" level=info msg="Start Longhorn Setting controller" controller=longhorn-setting node=k8s-worker-01
I0102 16:02:03.136231       1 shared_informer.go:197] Waiting for caches to sync for longhorn settings
I0102 16:02:03.136234       1 shared_informer.go:204] Caches are synced for longhorn settings
time="2021-01-02T16:02:03Z" level=info msg="Starting Longhorn instance manager controller"
I0102 16:02:03.136814       1 shared_informer.go:197] Waiting for caches to sync for longhorn instance manager
I0102 16:02:03.136817       1 shared_informer.go:204] Caches are synced for longhorn instance manager
time="2021-01-02T16:02:03Z" level=info msg="Start Longhorn share manager controller" controller=longhorn-share-manager node=k8s-worker-01
I0102 16:02:03.136945       1 shared_informer.go:197] Waiting for caches to sync for longhorn-share-manager-controller
I0102 16:02:03.136950       1 shared_informer.go:204] Caches are synced for longhorn-share-manager-controller
time="2021-01-02T16:02:03Z" level=info msg="Start kubernetes controller"
I0102 16:02:03.136984       1 shared_informer.go:197] Waiting for caches to sync for kubernetes
I0102 16:02:03.136988       1 shared_informer.go:204] Caches are synced for kubernetes
time="2021-01-02T16:02:03Z" level=info msg="Start Longhorn Kubernetes node controller"
I0102 16:02:03.137120       1 shared_informer.go:197] Waiting for caches to sync for longhorn kubernetes node
I0102 16:02:03.137152       1 shared_informer.go:204] Caches are synced for longhorn kubernetes node
time="2021-01-02T16:02:03Z" level=info msg="Start longhorn-kubernetes-pod-controller"
I0102 16:02:03.137255       1 shared_informer.go:197] Waiting for caches to sync for longhorn-kubernetes-pod-controller
I0102 16:02:03.137264       1 shared_informer.go:204] Caches are synced for longhorn-kubernetes-pod-controller
time="2021-01-02T16:02:03Z" level=info msg=Start controller=longhorn-kubernetes-configmap-controller node=k8s-worker-01
I0102 16:02:03.137328       1 shared_informer.go:197] Waiting for caches to sync for longhorn-kubernetes-configmap-controller
I0102 16:02:03.137413       1 shared_informer.go:204] Caches are synced for longhorn-kubernetes-configmap-controller
time="2021-01-02T16:02:05Z" level=debug msg="Updated setting default-engine-image to longhornio/longhorn-engine:v1.1.0"
time="2021-01-02T16:02:05Z" level=debug msg="Updated setting default-instance-manager-image to longhornio/longhorn-instance-manager:v1_20201216"
time="2021-01-02T16:02:05Z" level=debug msg="Updated setting default-share-manager-image to longhornio/longhorn-share-manager:v1_20201204"
time="2021-01-02T16:02:05Z" level=info msg="Event(v1.ObjectReference{Kind:\"Node\", Namespace:\"longhorn-system\", Name:\"k8s-worker-01\", UID:\"2e4c6aa1-35b6-461f-9edc-a18f14e0f6de\", APIVersion:\"longhorn.io/v1beta1\", ResourceVersion:\"4841\", FieldPath:\"\"}): type: 'Normal' reason: 'Schedulable' "
time="2021-01-02T16:02:05Z" level=info msg="Event(v1.ObjectReference{Kind:\"Node\", Namespace:\"longhorn-system\", Name:\"k8s-worker-01\", UID:\"2e4c6aa1-35b6-461f-9edc-a18f14e0f6de\", APIVersion:\"longhorn.io/v1beta1\", ResourceVersion:\"4841\", FieldPath:\"\"}): type: 'Warning' reason: 'Ready' Node k8s-worker-01 is down: the manager pod longhorn-manager-lsxqm is not running"
time="2021-01-02T16:02:05Z" level=debug msg="Prepare to create default instance manager instance-manager-e-bc8f8398, node: k8s-worker-01, default instance manager image: longhornio/longhorn-instance-manager:v1_20201216, type: engine"
time="2021-01-02T16:02:06Z" level=debug msg="Created engine image ei-cf743f9c (longhornio/longhorn-engine:v1.1.0)"
time="2021-01-02T16:02:06Z" level=debug msg="Waiting for engine image longhornio/longhorn-engine:v1.1.0 to be ready"
time="2021-01-02T16:02:06Z" level=debug msg="Instance Manager Controller k8s-worker-01 picked up instance-manager-e-bc8f8398"
time="2021-01-02T16:02:06Z" level=warning msg="Starts to clean up then recreates pod for instance manager instance-manager-e-bc8f8398 with state stopped"
time="2021-01-02T16:02:06Z" level=debug msg="Engine Image Controller k8s-worker-01 picked up ei-cf743f9c (longhornio/longhorn-engine:v1.1.0)"
time="2021-01-02T16:02:06Z" level=info msg="Created daemon set engine-image-ei-cf743f9c for engine image ei-cf743f9c (longhornio/longhorn-engine:v1.1.0)"
time="2021-01-02T16:02:07Z" level=info msg="Created instance manager pod instance-manager-e-bc8f8398 for instance manager instance-manager-e-bc8f8398"
time="2021-01-02T16:02:11Z" level=debug msg="Start monitoring instance manager instance-manager-e-bc8f8398"
time="2021-01-02T16:02:12Z" level=debug msg="Waiting for engine image longhornio/longhorn-engine:v1.1.0 to be ready"
time="2021-01-02T16:02:15Z" level=info msg="Event(v1.ObjectReference{Kind:\"EngineImage\", Namespace:\"longhorn-system\", Name:\"ei-cf743f9c\", UID:\"fd236d95-e478-479d-83db-47321750d593\", APIVersion:\"longhorn.io/v1beta1\", ResourceVersion:\"4876\", FieldPath:\"\"}): type: 'Normal' reason: 'ready' Engine image ei-cf743f9c (longhornio/longhorn-engine:v1.1.0) become ready"
time="2021-01-02T16:02:18Z" level=debug msg="Engine image longhornio/longhorn-engine:v1.1.0 is ready"
time="2021-01-02T16:02:18Z" level=info msg="Listening on 10.42.1.12:9500"
time="2021-01-02T16:02:19Z" level=info msg="Event(v1.ObjectReference{Kind:\"Node\", Namespace:\"longhorn-system\", Name:\"k8s-master-01\", UID:\"b992a27e-ddd5-42e0-8ac1-0c6ed18edfe8\", APIVersion:\"longhorn.io/v1beta1\", ResourceVersion:\"4919\", FieldPath:\"\"}): type: 'Normal' reason: 'Schedulable' "
time="2021-01-02T16:02:19Z" level=info msg="Event(v1.ObjectReference{Kind:\"Node\", Namespace:\"longhorn-system\", Name:\"k8s-master-01\", UID:\"b992a27e-ddd5-42e0-8ac1-0c6ed18edfe8\", APIVersion:\"longhorn.io/v1beta1\", ResourceVersion:\"4919\", FieldPath:\"\"}): type: 'Warning' reason: 'Ready' Node k8s-master-01 is down: the manager pod longhorn-manager-98lqf is not running"
time="2021-01-02T16:02:19Z" level=debug msg="Instance Manager Controller k8s-worker-01 picked up instance-manager-e-06171c88"
time="2021-01-02T16:02:24Z" level=info msg="Event(v1.ObjectReference{Kind:\"Node\", Namespace:\"longhorn-system\", Name:\"k8s-worker-01\", UID:\"2e4c6aa1-35b6-461f-9edc-a18f14e0f6de\", APIVersion:\"longhorn.io/v1beta1\", ResourceVersion:\"4851\", FieldPath:\"\"}): type: 'Normal' reason: 'Ready' Node k8s-worker-01 is ready"
time="2021-01-02T16:02:24Z" level=debug msg="Requeue longhorn-system/k8s-worker-01 due to conflict: Operation cannot be fulfilled on nodes.longhorn.io \"k8s-worker-01\": the object has been modified; please apply your changes to the latest version and try again"
time="2021-01-02T16:02:28Z" level=info msg="Event(v1.ObjectReference{Kind:\"Node\", Namespace:\"longhorn-system\", Name:\"k8s-master-01\", UID:\"b992a27e-ddd5-42e0-8ac1-0c6ed18edfe8\", APIVersion:\"longhorn.io/v1beta1\", ResourceVersion:\"4931\", FieldPath:\"\"}): type: 'Normal' reason: 'Ready' Node k8s-master-01 is ready"
time="2021-01-02T16:02:28Z" level=debug msg="Requeue longhorn-system/k8s-master-01 due to conflict: Operation cannot be fulfilled on nodes.longhorn.io \"k8s-master-01\": the object has been modified; please apply your changes to the latest version and try again"
time="2021-01-02T16:02:33Z" level=debug msg="Failed to check for the latest upgrade" controller=longhorn-setting error="Post \"https://longhorn-upgrade-responder.rancher.io/v1/checkupgrade\": dial tcp: i/o timeout" node=k8s-worker-01
```

**No** *Support Bundle* since the UI doesn't start.

**Environment:**
 - Longhorn version: `v1.0.0`
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: `k3s version v1.18.9+k3s1 (630bebf9)`
 - Node config
   - OS type and version: `k3os version v0.11.1`
   - CPU per node: 2
   - Memory per node: 4GB
   - Disk type(e.g. SSD/NVMe): HDD
   - Network bandwidth and latency between the nodes: `ESXI vSwitch / ~100ms`
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): `VMWare ESXi 6.0.0`

**Additional context**
Modified default settings for Helm:
```yaml
defaultSettings:
  allowRecurringJobWhileVolumeDetached: ~
  createDefaultDiskLabeledNodes:  true                  # Only create disk on nodes with special label
  defaultDataPath: /opt/longhorn                        # Path to store longhorn volume at
  defaultDataLocality: best-effort                      # Try to keep a storage replica on the same node as the workload
  replicaSoftAntiAffinity: ~
  storageOverProvisioningPercentage: ~
  storageMinimalAvailablePercentage: 10                 # Best practice for dedicated disk
  upgradeChecker: true                                  # Default
  defaultReplicaCount: 3                                # Default, best practice
  guaranteedEngineCPU: 0.25                             # Default, best practice
  defaultLonghornStaticStorageClass:  longhorn-static   # Default
  backupstorePollInterval: ~
  taintToleration: ~
  priorityClass: ~
  autoSalvage: ~
  autoDeletePodWhenVolumeDetachedUnexpectedly: ~
  disableSchedulingOnCordonedNode: true                 # Default, will not create storage on Master nodes
  replicaZoneSoftAntiAffinity: ~
  volumeAttachmentRecoveryPolicy: ~
  nodeDownPodDeletionPolicy: ~
  allowNodeDrainWithLastHealthyReplica: ~
  mkfsExt4Parameters: ~
  disableReplicaRebuild: ~
  replicaReplenishmentWaitInterval: ~
  disableRevisionCounter: ~
  systemManagedPodsImagePullPolicy: ~
  allowVolumeCreationWithDegradedAvailability: ~
  autoCleanupSystemGeneratedSnapshot: ~
```

**Comment by FabianNiehaus** [comment]
Update: Same issue arises when using no custom values.

```
NAME                                        READY   STATUS             RESTARTS   AGE
longhorn-driver-deployer-76c8fd69fb-j6k4t   0/1     Init:0/1           0          51s
instance-manager-e-482e1937                 1/1     Running            0          41s
instance-manager-r-f9e6b672                 1/1     Running            0          41s
engine-image-ei-cf743f9c-7nm7w              1/1     Running            0          42s
engine-image-ei-cf743f9c-s6bn2              1/1     Running            0          42s
longhorn-manager-6j6pf                      1/1     Running            2          51s
longhorn-manager-tbgq2                      1/1     Running            1          51s
instance-manager-e-96d2d065                 1/1     Running            0          26s
instance-manager-r-d1309d4a                 1/1     Running            0          25s
longhorn-ui-78547884d8-g6vqs                0/1     CrashLoopBackOff   2          51s
```

**Comment by shuo-wu** [comment]
1. What's the log/status of the longhorn-driver-deployer? I am not sure if other longhorn workloads are able to talk with longhorn manager pod. If No, that may be the root cause. You can check it by entering into one of the Longhorn workloads. e.g.:
```
~ kubectl -n longhorn-system exec -it <A Longhorn pod, e.g., engine image pods> -- bash
bash-5.0# curl -m 1 -s -o /dev/null -w "%{http_code}" http://longhorn-backend:9510/v1
200bash-5.0#
```
2. Notice that you are using HDD and the potential network latency is pretty high. Longhorn system cannot work stably under this environment.

**Comment by FabianNiehaus** [comment]
Okay, so I was investigating the issue and stumbled across the following:
1. After adding two more nodes to the cluster, most of the pods would start. However, new pods had an image pull error.
2. Looking at the event logs, the pods tried to pull images from a private registry. Turns out I misconfigured the `k3s` `registries.yaml` by overwriting the default `docker.io` entry.
3. After completely uninstalling and re-installing `Longhorn` via `Helm`, all pods are deployed properly.
4. I had an error in the configuration of my k3os nodes, leading to the Longhorn disks not being created. After fixing this, Longhorn works!
